### PR TITLE
Simple array and composition property values

### PR DESCRIPTION
**Pull request by jesperweber**
This PR contains two changes:

- Properties of simple array types like string and number are now supported and created using the Umbraco.Tags property type (we use the tags property type because it doesn't require predefined values on the datatype)
- Fixes property values for compositions. Prior to this PR properties on compositions was never given a value when migrating content. See example of types on below screenshot

image

### CODE REVIEW

File: src/Enterspeed/Enterspeed.Migrator/Enterspeed/PagesResolver.cs [hunk 1]
```diff
@@ -175,56 +175,20 @@ private void CreateArrayType(PageData pageData, JsonProperty jsonProperty, Enter
...
+                        {
+                            Name = "arrayObject",
+                            Alias = "arrayObject",
+                            Type = JsonValueKind.Object,
+                            Value = objectOfElement
+                        };
+
+                        // Add arrayitem directly to array property
+                        currentProperty.ChildProperties.Add(newArrayItem);
+                        MapData(pageData, element, newArrayItem);
```
Comment (jesperweber) [inline src/Enterspeed/Enterspeed.Migrator/Enterspeed/PagesResolver.cs:191]: Only values of type object should be created as child properties. Arrays values of string or numbers should not be created a child properties as it's just the value of the array.

File: src/Migrators/Umbraco/Umbraco.Migrator/Content/ContentBuilder.cs [hunk 2]
```diff
@@ -83,7 +77,11 @@ public void CreateProperties(List<EnterspeedPropertyType> properties, IContent c
 
                 if (property.ChildProperties.Any())
                 {
-                    CreateProperties(property.ChildProperties, contentToCreate, components, true);
+                    CreateProperties(property.ChildProperties, contentToCreate, components);
+                }
+                else
+                {
+                    AddValueToProperty(property, contentToCreate);
```
Comment (jesperweber) [inline src/Migrators/Umbraco/Umbraco.Migrator/Content/ContentBuilder.cs:84]: Even if we are traversing we should add values else compositions with child properties will never get a value. Have I missed any cases where we shouldn't add a value if we are traversing like before?

File: src/Migrators/Umbraco/Umbraco.Migrator/DocumentTypes/DocumentTypeBuilder.cs [hunk 3]
```diff
@@ -225,6 +225,13 @@ private void AddProperties(EnterspeedPropertyType enterspeedProperty, IContentTy
                     break;
                 case JsonValueKind.Array:
                     // Check if we are a component or simple type of array
```
Comment (jesperweber) [inline src/Migrators/Umbraco/Umbraco.Migrator/DocumentTypes/DocumentTypeBuilder.cs:227]: As the comment also mentioned before we need to check the type of the elements in the array, to see if it's a complex object or a simple type

Comment (MHove89) [reply src/Migrators/Umbraco/Umbraco.Migrator/DocumentTypes/DocumentTypeBuilder.cs:227]: This makes sense

**Review by MHove89 — APPROVED** [review]

**Pull request merged**